src/ngraph/runtime/gpu/gpu_cuda_kernel_builder.cpp

```diff
@@ -242,6 +242,66 @@ void runtime::gpu::CudaKernelBuilder::get_reverse_op(codegen::CodeWriter& writer
     writer.block_end();
 }
 
+void runtime::gpu::CudaKernelBuilder::get_reduce_window_op(
+    codegen::CodeWriter& writer,
+    const std::string& name,
+    const std::string& op,
+    const std::vector<std::string>& data_types,
+    const size_t rank)
+{
+    writer << "extern \"C\" __global__ void cuda_" << name << "(" << data_types[0] << "* in, "
+           << data_types[1] << "* out, size_t* input_strides, size_t* output_shape, size_t* "
```

Comment:
If you switch to GPUShape above we can declare all of these as arrays as `int` or `uint32_t` and avoid the performance loss of 64 bit words. Consider switching `size_t` throughout the kernel to 32 bits.